nodebuilder/tests/prune_test.go

```diff
@@ -94,14 +101,14 @@ func TestArchivalBlobSync(t *testing.T) {
 		root   share.DataHash
 	}
 
+	const wantBlobs = 10
 	archivalBlobs := make([]*archivalBlob, 0)
-	i := 1
-	for {
+
+	for i := 1; len(archivalBlobs) < wantBlobs; i++ {
```

Comment:
We just need to collect `wantBlobs` here, `i` is a simple increment that can be done in a loop declaration.